[2024-06-17T04:15:06.226+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-17T04:15:06.273+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Air_pollution_data_pipeline.generate_summary manual__2024-06-17T04:12:37+00:00 [queued]>
[2024-06-17T04:15:06.286+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Air_pollution_data_pipeline.generate_summary manual__2024-06-17T04:12:37+00:00 [queued]>
[2024-06-17T04:15:06.287+0000] {taskinstance.py:2306} INFO - Starting attempt 2 of 6
[2024-06-17T04:15:06.307+0000] {taskinstance.py:2330} INFO - Executing <Task(_PythonDecoratedOperator): generate_summary> on 2024-06-17 04:12:37+00:00
[2024-06-17T04:15:06.316+0000] {standard_task_runner.py:63} INFO - Started process 1220 to run task
[2024-06-17T04:15:06.322+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'Air_pollution_data_pipeline', 'generate_summary', 'manual__2024-06-17T04:12:37+00:00', '--job-id', '210', '--raw', '--subdir', 'DAGS_FOLDER/main.py', '--cfg-path', '/tmp/tmph8r25_d8']
[2024-06-17T04:15:06.326+0000] {standard_task_runner.py:91} INFO - Job 210: Subtask generate_summary
[2024-06-17T04:15:06.405+0000] {task_command.py:426} INFO - Running <TaskInstance: Air_pollution_data_pipeline.generate_summary manual__2024-06-17T04:12:37+00:00 [running]> on host fa4d239664b0
[2024-06-17T04:15:06.637+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='khalid' AIRFLOW_CTX_DAG_ID='Air_pollution_data_pipeline' AIRFLOW_CTX_TASK_ID='generate_summary' AIRFLOW_CTX_EXECUTION_DATE='2024-06-17T04:12:37+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-06-17T04:12:37+00:00'
[2024-06-17T04:15:06.639+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-17T04:15:06.642+0000] {summary_analysis.py:13} INFO - starting summary statistics of pollutants emission
[2024-06-17T04:15:06.656+0000] {logging_mixin.py:188} WARNING - /opt/***/dags/statistical_analysis/summary_analysis.py:16 FutureWarning: Passing literal json to 'read_json' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
[2024-06-17T04:15:06.658+0000] {summary_analysis.py:26} ERROR - error calculating pollutants summary statistics
Traceback (most recent call last):
  File "/opt/airflow/dags/statistical_analysis/summary_analysis.py", line 16, in generate_summary
    val_data = pd.read_json(val_data)
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 815, in read_json
    return json_reader.read()
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 1025, in read
    obj = self._get_object_parser(self.data)
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 1051, in _get_object_parser
    obj = FrameParser(json, **kwargs).parse()
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 1187, in parse
    self._parse()
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 1403, in _parse
    ujson_loads(json, precise_float=self.precise_float), dtype=None
ValueError: Expected object or value
[2024-06-17T04:15:06.661+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-17T04:15:06.663+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/decorators/base.py", line 265, in execute
    return_value = super().execute(context)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/statistical_analysis/summary_analysis.py", line 16, in generate_summary
    val_data = pd.read_json(val_data)
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 815, in read_json
    return json_reader.read()
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 1025, in read
    obj = self._get_object_parser(self.data)
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 1051, in _get_object_parser
    obj = FrameParser(json, **kwargs).parse()
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 1187, in parse
    self._parse()
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 1403, in _parse
    ujson_loads(json, precise_float=self.precise_float), dtype=None
ValueError: Expected object or value
[2024-06-17T04:15:06.681+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=Air_pollution_data_pipeline, task_id=generate_summary, run_id=manual__2024-06-17T04:12:37+00:00, execution_date=20240617T041237, start_date=20240617T041506, end_date=20240617T041506
[2024-06-17T04:15:06.707+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 210 for task generate_summary (Expected object or value; 1220)
[2024-06-17T04:15:06.738+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-17T04:15:06.761+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
